drivers/input/input_nunchuk.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2024 Bootlin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nintendo_nunchuk
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/timing/timing.h>
+
+LOG_MODULE_REGISTER(input_nunchuk, CONFIG_INPUT_LOG_LEVEL);
+
+struct nunchuk_config {
+	struct i2c_dt_spec i2c_bus;
+	k_timeout_t polling_interval;
+};
+
+struct nunchuk_data {
+	const struct device *dev;
+	uint8_t joystick_x;
+	uint8_t joystick_y;
+	bool button_c;
+	bool button_z;
+	struct k_work_delayable work;
+};
+
+int nunchuk_read_registers(const struct device *dev, uint8_t *buffer)
+{
+	const struct nunchuk_config *cfg = dev->config;
+	int ret;
+	uint8_t value = 0;
+
+	ret = i2c_write_dt(&cfg->i2c_bus, &value, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	k_msleep(10);
+	ret = i2c_read_dt(&cfg->i2c_bus, buffer, 6);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+void nunchuk_poll(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct nunchuk_data *data = CONTAINER_OF(dwork, struct nunchuk_data, work);
+	const struct device *dev = data->dev;
+	const struct nunchuk_config *cfg = dev->config;
+	uint8_t joystick_x, joystick_y;
+	bool button_c, button_z;
+	uint8_t buffer[6];
+	int ret;
+
+	nunchuk_read_registers(dev, buffer);
+
+	joystick_x = buffer[0];
+	if (joystick_x != data->joystick_x) {
+		data->joystick_x = joystick_x;
+		ret = input_report_abs(dev, INPUT_ABS_X, data->joystick_x, false, K_FOREVER);
+	}
+
+	joystick_y = buffer[1];
+	if (joystick_y != data->joystick_y) {
+		data->joystick_y = joystick_y;
+		ret = input_report_abs(dev, INPUT_ABS_Y, data->joystick_y, false, K_FOREVER);
```

Comment:
the sync bit logic here a is a bit tricky, basically this should do either
- report x+sync
- report y+sync
- report x and then y+sync

the buttons are probably fine to stay both with "sync" set all the time

so there's two options:
1. you always set the sync bit here and implement a logic to set it for x only if y did not change, or
2. you always send either both x and y or not, that's probably the easiest way, something like, up to you

Reply:
I did the first solution, it is cleaner I think